borg/_chunker.c

```diff
@@ -162,9 +163,16 @@ chunker_fill(Chunker *c)
         // complete cache with data that we only read once and (due to cache
         // size limit) kick out data from the cache that might be still useful
         // for the OS or other processes.
-        if (length > 0) {
-            posix_fadvise(c->fh, offset, length, POSIX_FADV_DONTNEED);
-        }
+        // We rollback the initial offset back to the start of the page,
+        // to avoid it not being truncated as a partial page request.
```

Comment:
how about adding here:
mask = getpagesize() - 1

and reusing it at 2 places below?